### PR TITLE
css and html bug fixes and style updates

### PR DESCRIPTION
**Pull request by bulam**
**Description of changeset:** 
**CSS & HTML bug fixes & updates:**
* CSS updates to match current Airbnb design elements. Mostly color scheme and border radius updates. Can be previewed here https://github.com/airbnb/knowledge-repo/blob/0931185c73d62800b49152ea27cef5527251990d/README.md
* HTML fix to the “Create a new post” page to fix the bug that was preventing center aligning
* Add missing div tag that was causing a rendering bug.
* Remove extra apostrophe in the HTML in the create a post page.
* Fixed the web editor missing placeholder text

**Copy updates in the UI to clarify functionalities and guide best practices**
* Updated the “Create a Post” page to describe in more detail the options to add new posts
* Added placeholder text for some fields in the add a post form (post path, post image thumbnail) to guide best practices or to guide usage
* Renamed the Web Proxy field to “Linked Post URL” for clarity
* Changed the “Write a Post!” button to “Add a Post!” reflecting the fact that you can add linked posts as well
* Changed the “Web Proxy” to “Linked Post” to make this option clearer.

* Also updated the screenshots in the readme to show the updated UI and copy
* Added more screenshots to the readme

**Test Plan:** 
* I tested the CSS and HTML fixes and updates in Chrome, Firefox and Safari in both desktop and mobile web views.

### CODE REVIEW

File: knowledge_repo/app/static/css/pages/base.css [hunk 1]
```diff
@@ -4,22 +4,29 @@
 }
 
 .navbar {
-    background-color: #565a5c;
-    border-color: #565a5c;
-    color: #fff;
-    box-shadow: 0px 1px 5px rgba(0,0,0,0.5);
+    background-color: #fff;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:7]: let's use `white` instead of `#fff`

File: knowledge_repo/app/static/css/pages/base.css [hunk 2]
```diff
@@ -4,22 +4,29 @@
...
+    border-color: #d2d2d2;
+    color: #000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:9]: let's use `black` instead of `#000`

File: knowledge_repo/app/static/css/pages/base.css [hunk 3]
```diff
@@ -4,22 +4,29 @@
...
+    box-shadow: 0px 1px 3px rgba(0, 0, 0, 0.15);
     z-index: 10000;
 }
 .navbar-inverse .navbar-nav>li>a {
-  color: #fff;
+  color: #000000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:14]: use ` black` here

File: knowledge_repo/app/static/css/pages/base.css [hunk 4]
```diff
@@ -4,22 +4,29 @@
...
 }
+
+.navbar-inverse .navbar-nav>li>a:focus, .navbar-inverse .navbar-nav>li>a:hover {
+  color: #000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:18]: use ` black` here

File: knowledge_repo/app/static/css/pages/base.css [hunk 5]
```diff
@@ -4,22 +4,29 @@
...
+}
+
 .navbar .navbar-brand {
   float: none;
-  color: #fff;
+  color: #000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:23]: use ` black` here

File: knowledge_repo/app/static/css/pages/base.css [hunk 6]
```diff
@@ -4,22 +4,29 @@
...
   display: flex;
   align-items: center;
   padding-left: 0px;
   padding-right: 0px;
+  font-weight: bolder;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:28]: is `bolder` what you want here? That's saying this text should be bolder than the parent's element. It might be better to set this to `700` instead

File: knowledge_repo/app/static/css/pages/base.css [hunk 7]
```diff
@@ -31,6 +38,20 @@
...
+.form-control {
+  border-radius: 21px;
+  box-shadow: 0 0px 0px rgba(0,0,0,0);
+  border: 1px solid #e2ddddcf;
+  padding: 8px 14px;
+}
+
+.btn {
+  border-radius: 21px;
+  font-weight: 450;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:52]: `450` probably isn't a real font weight here (ie the font files likely don't define a `450`). You probably want `400` or `500`

File: knowledge_repo/app/static/css/pages/base.css [hunk 8]
```diff
@@ -31,6 +38,20 @@
         display: flex!important;
     }
 }
+
+/* Nav bar search form design */
+.form-control {
+  border-radius: 21px;
+  box-shadow: 0 0px 0px rgba(0,0,0,0);
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:45]: I think this can be `0 0 0 black` instead? But actually, if this is unsetting box-shadow, you might be able to set it to `unset` instead

File: knowledge_repo/app/static/css/pages/base.css [hunk 9]
```diff
@@ -47,44 +68,73 @@
 }
 #navbar .navbar-form .input-group #searchbar {
     width: 100%;
-    background: #999;
-    color: #fff;
+    background: #ececec;
+    color: #000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:72]: use `black`

File: knowledge_repo/app/static/css/pages/base.css [hunk 10]
```diff
@@ -47,44 +68,73 @@
...
+    opacity: 1.0;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:73]: `1` instead of `1.0`

File: knowledge_repo/app/static/css/pages/base.css [hunk 11]
```diff
@@ -47,44 +68,73 @@
...
     border-color: whitesmoke;
+    background-color: #ececec;
+    border-color: #ececec;
 }
 #navbar .navbar-form .input-group #searchbar::-webkit-input-placeholder { /* Chrome */
-  color: #fff;
+  color: #000;
   opacity: 0.5;
-  font-style: italic;
+  /*font-style: italic;*/
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:91]: instead of commenting this out, can you delete the line instead?

File: knowledge_repo/app/static/css/pages/base.css [hunk 12]
```diff
@@ -47,44 +68,73 @@
...
 #navbar .navbar-form .input-group #searchbar:focus {
     background: #aaa;
     box-shadow: none;
     transition: background 0.3s;
+    background: #ececec;
 }
 #navbar .navbar-form .btn-default {
     border-color: whitesmoke;
+    background-color: #ececec;
+    border-color: #ececec;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:86]: if you're resetting the `border-color`, you can delete line 84 which defines it already

File: knowledge_repo/app/static/css/pages/base.css [hunk 13]
```diff
@@ -47,44 +68,73 @@
...
 }
 #navbar .navbar-form .input-group #searchbar::-webkit-input-placeholder { /* Chrome */
-  color: #fff;
+  color: #000;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:89]: use `black`. I'll stop commenting this from here on out, but please update the remaining cases

File: knowledge_repo/app/static/css/pages/base.css [hunk 14]
```diff
@@ -47,44 +68,73 @@
...
+}
+
+
+.navbar-inverse .navbar-brand:focus, .navbar-inverse .navbar-brand:hover {
+    color: #000;
 }
+
+/* navbar click background color*/
+.navbar-inverse .navbar-nav>.open>a, .navbar-inverse .navbar-nav>.open>a:focus, .navbar-inverse .navbar-nav>.open>a:hover {
+  background-color: #ffffff;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:121]: use `white` (i'll stop commenting this from here on out)

File: knowledge_repo/app/static/css/pages/base.css [hunk 15]
```diff
@@ -269,19 +324,23 @@
...
 
 .panel-header {
-    color: #565a5c;
-    font-size: 16px;
+    color: #232323;
+    font-size: 18px;
     padding-top: 14px;
     padding-bottom: 14px;
-    border-bottom: 1px solid #dce0e0;
+    /*border-bottom: 1px solid #dce0e0;*/
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:341]: delete line instead of commenting

File: knowledge_repo/app/static/css/pages/base.css [hunk 16]
```diff
@@ -297,18 +356,19 @@
     position: relative;
     margin: 0;
     padding: 20px;
-    border-top: 1px solid #dce0e0;
+    /*border-top: 1px solid #dce0e0;*/
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:359]: delete line instead of commenting

File: knowledge_repo/app/static/css/pages/base.css [hunk 17]
```diff
@@ -360,7 +420,18 @@ img {
 
 .label-primary,
 .label-default,
-.label-subscribed,
+.label-subscribed {
+  font-size: 14px;
+  border: 1px solid;
+  border-radius: 15px;
+  border-color: ##00A699;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:427]: remove the extra `#`

File: knowledge_repo/app/static/css/pages/base.css [hunk 18]
```diff
@@ -360,7 +420,18 @@ img {
 
 .label-primary,
 .label-default,
-.label-subscribed,
+.label-subscribed {
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:423]: what's the purpose of this change? If it's simply to change the color of the subscribed label, you can keep this all the same, then below it add:
```css
.label-subscribed {
  border-color: #00A699
}
```

File: knowledge_repo/app/static/css/pages/base.css [hunk 19]
```diff
@@ -491,9 +562,9 @@ a.label:focus, a.label:hover {
 .btn-primary:focus,
 .btn-primary:active {
   color:white;
-  background-color: #9CA299;
-  border-color: #9CA299;
-  opacity: 0.5;
+  background-color: #06c6b6;
+  border-color: #06c6b6;
+  opacity: 1.0;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:567]: `1` instead of `1.0`

File: knowledge_repo/app/static/css/pages/base.css [hunk 20]
```diff
@@ -600,11 +671,21 @@ span.index-view-name {
...
+}
+
 .btn-active:hover {
-  background-color: #00a699;
-  opacity: 0.8;
+  background-color: #06c6b6;
+  opacity: 1;
 }
 
+/*.btn-active .index-view-name:hover {
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:684]: delete instead of comment

File: knowledge_repo/app/static/css/pages/base.css [hunk 21]
```diff
@@ -801,3 +905,27 @@ border-color: #11b6aa;
...
+h5 {
+  margin-block-start: 11px;
+  margin-block-end: 0px;
+}
+
+
+
+.select2-container .select2-selection--multiple {
+    background-color: white;
+    border: 1px solid #e2ddddcf!important;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:918]: generally using `!important` is not recommended. Is there another way you can set these attributes?

File: knowledge_repo/app/static/css/pages/index-feed.css [hunk 22]
```diff
@@ -31,10 +33,10 @@
     width: 200px;
     position: relative;
     vertical-align: center;
-    background: #fff;
-    border: 1px solid #bbb;
+    /*background: #fff;*/
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/index-feed.css:36]: delete instead of commenting

File: knowledge_repo/app/templates/auth-login-form.html [hunk 23]
```diff
@@ -27,6 +28,10 @@
 .login-form > input[type=submit] {
     margin-bottom: 0px;
 }
+
+button, input, select, textarea {
+    font-weight: 450;
```
Comment (etr2460) [inline knowledge_repo/app/templates/auth-login-form.html:33]: i don't think `450` has meaning, this should probably be `400` or `500`

File: knowledge_repo/app/templates/base.html [hunk 24]
```diff
@@ -94,7 +94,7 @@
 
             <!-- "Write a Post" button -->
```
Comment (etr2460) [inline knowledge_repo/app/templates/base.html:95]: Let's update the comment here too if we're switching to `Add`

File: knowledge_repo/app/templates/create-knowledge.html [hunk 25]
```diff
@@ -7,15 +7,26 @@
 {% block content %}
 <center>
   <br>
-  <h3> Select format to download a knowledge stub. </h3>
+  <h3> Select from the options below to add a new post </h3>
```
Comment (etr2460) [inline knowledge_repo/app/templates/create-knowledge.html:10]: maybe: `Select an option below to add a new post`?

File: knowledge_repo/app/templates/post_editor_base.html [hunk 26]
```diff
@@ -8,12 +8,10 @@
...
 
 {% block content %}
-    {{ super() }}
-
-    <div class="panel-webpost container">
-      <div class="col-md-12"  id="#header-panel">
-        <h3> {{ path if path else 'New Knowledge Post' }}
-
+  {{ super() }}
+  <div class="col-lg-14">
```
Comment (etr2460) [inline knowledge_repo/app/templates/post_editor_base.html:12]: Are there non formatting changes in this file other than this line? It's tough for me to tell

File: knowledge_repo/app/static/css/pages/base.css (hunk 1 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:7]: 👍

File: knowledge_repo/app/static/css/pages/base.css (hunk 2 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:9]: 👍

File: knowledge_repo/app/templates/post_editor_base.html (hunk 26 above)
Comment (bulam) [reply knowledge_repo/app/templates/post_editor_base.html:12]: Yes, besides this line, it's mostly formatting changes:
- Indenting the code for readability
- Added a missing div tag

Changing this line from 12 to 14 fixed the alignment of the change. When set to 12, the form was aligned to the right of the page.

File: knowledge_repo/app/static/css/pages/base.css (hunk 14 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:121]: 👍 I'll update all of them

**Comment by bulam** [comment]
@etr2460 Thanks for all the comments and great suggestions. All of them make sense. I'll implement in the next few commits.

File: knowledge_repo/app/static/css/pages/base.css (hunk 21 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:905]: I actually tried set it multiple ways and couldn't get it work, so I ended up using !important.

File: knowledge_repo/app/static/css/pages/base.css (hunk 8 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css]: `unset` worked!

File: knowledge_repo/app/static/css/pages/base.css (hunk 18 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css]: weird. I tried exactly this just and the button radius and coloring become totally different

File: knowledge_repo/app/static/css/pages/base.css (hunk 12 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:85]: 👍

File: knowledge_repo/app/static/css/pages/base.css (hunk 6 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css]: 👍

File: knowledge_repo/app/static/css/pages/base.css (hunk 7 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css]: 👍

File: knowledge_repo/app/templates/create-knowledge.html (hunk 25 above)
Comment (bulam) [reply knowledge_repo/app/templates/create-knowledge.html]: 👍

File: knowledge_repo/app/templates/base.html (hunk 24 above)
Comment (bulam) [reply knowledge_repo/app/templates/base.html]: 👍

File: knowledge_repo/app/templates/auth-login-form.html (hunk 23 above)
Comment (bulam) [reply knowledge_repo/app/templates/auth-login-form.html]: 👍

**Comment by bulam** [comment]
@etr2460 I implemented the suggested changes and implemented a few more changes.

**Comment by bulam** [comment]
@etr2460 I chatted with Ken and incorporated his suggestions into the design.

I also found the commenting feature in the webpost submission form is not working, so I disabled the feature for now and filed an issue here https://github.com/airbnb/knowledge-repo/issues/559

File: knowledge_repo/app/static/css/pages/base.css [hunk 27]
```diff
@@ -733,7 +817,10 @@ border-color: #11b6aa;
 }
 
 .webeditor-form > label {
-  font-weight: 200;
+    font-weight: 450;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:820]: can you use either 400 or 500 here? again, i don't think 450 is valid

File: knowledge_repo/app/static/css/pages/base.css [hunk 28]
```diff
@@ -801,3 +911,34 @@ border-color: #11b6aa;
   height:70px; /* fixed header height*/
   margin:-70px 0 0; /* negative fixed header height */
 }
+
+h5 {
+  margin-block-start: 11px;
+  margin-block-end: 0px;
+}
+
+
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:920]: remove extra 2 blank lines

File: knowledge_repo/app/static/css/pages/base.css [hunk 29]
```diff
@@ -801,3 +911,34 @@ border-color: #11b6aa;
...
+    font-weight: 500!important;
+    color: #333333!important;
+}
+
+button, input, select, textarea {
+    border-style: none;
+    border: 1px solid currentColor;
+    border-radius: 8px;
+    border-color: black;
+}
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/base.css:944]: make sure there's a newline at the end of the file

File: knowledge_repo/app/static/css/pages/base.css (hunk 21 above)
Comment (etr2460) [reply knowledge_repo/app/static/css/pages/base.css:924]: what exactly are you trying to accomplish here? Maybe you can add a new css class to the element you're trying to style.

File: knowledge_repo/app/static/css/pages/index-feed.css [hunk 30]
```diff
@@ -44,11 +43,10 @@
     right: 0;
     bottom: 0;
     left: 0;
-    max-width:150px;
-    max-height: 150px;
-    height:auto;
-    width:auto;
+    max-height:200px;
+    width:200px;
```
Comment (etr2460) [inline knowledge_repo/app/static/css/pages/index-feed.css:47]: nit: add spaces after the colon

File: knowledge_repo/app/templates/post_editor_base.html [hunk 31]
```diff
@@ -560,7 +563,7 @@ <h2> {{ comments | length }} Comments </h2>
   });
 </script>
 <script type="text/javascript"
-  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.8/MathJax.js">
+  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js">
```
Comment (etr2460) [inline knowledge_repo/app/templates/post_editor_base.html:566]: is this an intentional change to downgrade this library?

**Comment by naoyak** [comment]
> @etr2460 I chatted with Ken and incorporated his suggestions into the design.
> 
> I also found the commenting feature in the webpost submission form is not working, so I disabled the feature for now and filed an issue here #559

I noticed this too but is this just an issue in the debug server? The production instance seems to work...

File: knowledge_repo/app/templates/post_editor_base.html (hunk 31 above)
Comment (bulam) [reply knowledge_repo/app/templates/post_editor_base.html:566]: sorry no. I'll update back

File: knowledge_repo/app/static/css/pages/base.css (hunk 21 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css:922]: These are styling the interactive author and tags entry fields that have autosuggest functionality. The styles are to have them match with the rest of the UI. These come from the `select2` library and I found I couldn't get the style customization to apply unless I marked them as important.

**Comment by bulam** [comment]
> > I also found the commenting feature in the webpost submission form is not working, so I disabled the feature for now and filed an issue here #559
> 
> I noticed this too but is this just an issue in the debug server? The production instance seems to work...

Chatted with Naoya and found the comment feature in the post review flow is not working in prod either

File: knowledge_repo/app/static/css/pages/base.css (hunk 21 above)
Comment (bulam) [reply knowledge_repo/app/static/css/pages/base.css]: Finally got this to work. Referred to the class by a parent ID

**Comment by bulam** [comment]
Was able to make the styling work without using the important. Does this look good for merging? @etr2460
